src/Umbraco.Core/StringExtensions.cs

```diff
@@ -1465,10 +1470,7 @@ internal static string ToValidXmlString(this string text)
         /// <returns></returns>
         internal static Guid ToGuid(this string text)
         {
-            var md5 = MD5.Create();
-            byte[] myStringBytes = Encoding.ASCII.GetBytes(text);
-            byte[] hash = md5.ComputeHash(myStringBytes);
-            return new Guid(hash);
+            return new Guid(text.GenerateHash());

```

Comment:
According to this SHA1 should work, but I will need to edit this to truncate the hash to 128 bits

https://stackoverflow.com/a/6248764/32908